Automattic/wp-calypso
# Framework/CPT: Track current page title in Redux state

This pull request seeks to implement new state behaviors and bindings for tracking the current page title in the global application state. Specifically, it includes a new action and reducer pair, a `<QueryTitle />` query component for assigning title, and bindings necessary to interoperate with the existing Flux-based [`screen-title` store](https://github.com/Automattic/wp-calypso/tree/master/client/lib/screen-title).

__Background:__

Custom post type labels can be localized. The localized label should be used in the `<title>` element of the page. The labels are accessed on the post types tracked in global application state. Rather than subscribe to state from the controller, it seems more reasonable to adopt our newer [query component pattern](https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md#query-components) in assigning title to state via a rendered component at the top of the section render hierarchy.

__Implementation notes:__

Since many sections continue to rely on the `screen-title` Flux store, the title tracked in state may become inaccurate as the user navigates through the application, even with these changes. For example, after navigating from Types listing to stats, `state.ui.title` will continue to be assigned as "Custom Post Type". This issue will be alleviated if more sections adopt the Redux state approach introduced here.

__Testing instructions:__

Ensure that the title is updated correctly when navigating to the Custom Post Types route for a site. You must navigate to the URL directly in your development environment, as there are no navigable links yet available.

1. Navigate to [Calypso custom post types listing](http://calypso.localhost:3000/types/jetpack-portfolio) (e.g. portfolio)
2. Select a site
3. Note that the tab title reflects both "Custom Post Types" and the title of the selected site
4. Click Switch Sites in the sidebar navigation
5. Select a site
6. Note that the tab title reflects both "Custom Post Types" and the title of the selected site
7. Navigate to another page in Calypso
8. Note that the tab title reflects the section you chose
